libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -3847,14 +3850,28 @@ CK_DECLARE_FUNCTION( CK_RV, C_GenerateRandom )( CK_SESSION_HANDLE xSession,
 
     if( xResult == CKR_OK )
     {
-        lMbedResult = mbedtls_ctr_drbg_random( &xP11Context.xMbedDrbgCtx, pucRandomData, ulRandomLen );
-
-        if( lMbedResult != 0 )
+        while( ( lLoops < ENTROPY_MAX_LOOPS ) && ( totalBytes < ulRandomLen ) )
```

Comment:
Is there any blocking behavior here?  Is there another interface that permits a process to block until enough entropy is available?

Reply:
since entropy calls are port specific, I don't think generally there is a entropy available callback that I can rely on, and I would be dependent on the port implementation to decide when it wants to yield.

Reply:
After offline conversation, have added a  delay to give hardware time to gather more entropy